File: ProcessPayroll/Section A/Employee.cs
namespace SectionA;
public class Employee {
  public string Nric { get; set; }
  public string FullName { get; set; }
  public string Salutation { get; set; }
  public DateTime StartDate { get; set; }
  public string Designation { get; set; }
  public string Department { get; set; }
  public string MobileNo { get; set; }
  public string HireType { get; set; }
  public double Salary { get; set; }
  public double MonthlyPayout { get; set; }

  public Employee(string record) {
    string[] data = record.Split('|');
    Nric = data[0];
    FullName = data[1];
    Salutation = data[2];
    StartDate = DateTime.Parse(data[3]);
    Designation = data[4];
    Department = data[5];
    MobileNo = data[6];
    HireType = data[7];
    Salary = double.Parse(data[8]);
  }

  public string FormatInfoForCorpAdmin() => $"{FullName}, {Designation}, {Department}";
  public string FormatInfoForProcurement() => $"{Designation}, {FullName}, {MobileNo}, {Designation}, {Department}";
  public string FormatInfoForITDepartment() => $"{Nric}, {FullName}, {StartDate.ToString(@"dd/MM/yyyy")}, {Department}, {MobileNo}";
  public override string ToString() => $"{Nric}|{FullName}|{Salutation}|{StartDate.ToString(@"dd/MM/yyyy")}|{Department}|{Designation}|{MobileNo}|{HireType}|{Salary}";
}

File: ProcessPayroll/Section C/Employee.py
from datetime import datetime


class Employee:
    def __init__(self, record):
        data = record.split('|')
        self.__nric = data[0]
        self.__full_name = data[1]
        self.__salutation = data[2]
        self.__start_date = datetime(*[int(i) for i in data[3][:11].split("/")[::-1]])
        self.__designation = data[4]
        self.__department = data[5]
        self.__mobile_no = int(data[6])
        self.__hire_type = data[7]
        self.__salary = int(data[8])

    def get_start_year(self): return self.__start_date.year
    def get_salary(self): return self.__salary
    def get_hire_type(self): return self.__hire_type
    def set_salary_by_multiplier(self, multiplier): 
        self.__salary *= multiplier
        print(self)
        return self

    def __str__(self): return f"{self.__full_name} ({self.__nric})\n{self.__designation}\n{self.__hire_type} Payout: ${self.__salary:.0f}\n------------------------"
 